### ISSUE TITLE
Walk-through: Using an Entity Framework 6 EDMX file with .NET Core

### ISSUE THREAD

**Issue by ErikEJ** [ErikEJ/erikej.github.io#12]
https://erikej.github.io/ef/dotnetcore/2020/06/15/ef6-use-edmx-dotnetcore.html

**Comment by amosialek** [comment]
Does it still work with .Net6?

**Comment by ErikEJ** [comment]
@amosialek Sure

**Comment by audacity76** [comment]
Thanks Erik for this walk-through!

I found out, it's easier (for me) to just add a .net framework project besides the .net 6 (core) project and include the edmx and tt files. That way I can remove / add entities in the .net framework project and don't need to think about linking them because .net 6 projects includes them automatically. Also worth noting, the .net framework project doesn't need to build, so it can be excluded in the build configuration. 

Also, if both projects are in the same solution one big issue can be the order in the solution. Docs:

> Warning: make sure the old style (i.e. non-SDK-style) .NET Framework project defining the "real" .edmx file comes before the project defining the link inside the .sln file. Otherwise, when you open the .edmx file in the designer, you see the error message "The Entity Framework is not available in the target framework currently specified for the project. You can change the target framework of the project or edit the model in the XmlEditor".

**Comment by BlackbirdSQL** [comment]
> I found out, it's easier (for me) to just add a .net framework project besides the .net 6 (core) project and include the edmx and tt files. That way I can remove / add entities in the .net framework project and don't need to think about linking them because .net 6 projects includes them automatically.

@audacity76 Hi Sven. Can you describe how you did this? I've tried various ways of including the tt's in the "client" .NET project but but I don't get classes automatically added for any subsequent entities I add to the "provider" .NetFramework project.
Thanks, Greg

**Comment by audacity76** [comment]
@BlackbirdSQL Hi Greg, if the projects share the same folders the files get automatically added in .NET. That's the normal behavior since .NET core project structure. It's the case that you'll need to specify the files that you don't want to include in your project. So, have a look at your client .NET project file if there is any exclusion added.

**Comment by mohd-anwar1** [comment]
Hi @ErikEJ . I'm getting the below error in DotNet 6.
"The ADO.NET provider with invariant name 'Oracle.ManagedDataAccess.Client' is either not registered in the machine or application config file, or could not be loaded. See the inner exception for details". Can I please have the completed version of WindowsFormsApplication1.zip.

**Comment by ErikEJ** [comment]
@mohd-anwar1 Have you installed the Oracle Visual Studio Tools?

**Comment by mohd-anwar1** [comment]
@ErikEJ  Yes. I have installed Oracle.VsDevTools.17.0.vsix for VS 2022 and ODT extension in VS 2022 shows enabled.

**Comment by ErikEJ** [comment]
@mohd-anwar1 please contact Oracle support

**Comment by gasand** [comment]
Worked perfectly for me! Thanks @ErikEJ I also linked the partial classes I use to extend EF Entities and it's working great. I think this might be a good approach to start migrating a .net 4.8 solution to .net core. Without the hassle of redesigning the DAL.

**Comment by jbogard** [comment]
Just did this with .NET 8 and can confirm it continues to work there too

**Comment by ErikEJ** [comment]
@jbogard Thanks for confirming. Do you use thenew Microsoft.EntityFramework.SqlServer package that I authored?

**Comment by jbogard** [comment]
Not yet - it's a whole separate effort we have to move to Azure.Identity, managed identities etc. Baby steps!

**Comment by ErikEJ** [comment]
@jbogard Reach out if you get blocked in any way!
